api-server/migrations/20230416142015-add-columns-for-tags.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    return queryInterface.sequelize.transaction(t => {
+      return Promise.all([
+        queryInterface.addColumn(
+          'datetime_venue',
+          'category',
+          {
+            type: Sequelize.STRING,
+            allowNull: true
+          }, { transaction: t }),
+        queryInterface.addColumn(
+          'events',
+          'category',
+          {
+            type: Sequelize.STRING,
+            allowNull: true
+          }, { transaction: t }),
+        queryInterface.addColumn(
+          'events',
+          'condition',
+          {
+            type: Sequelize.ARRAY(Sequelize.STRING),
+            allowNull: true
+          }, { transaction: t }),
+        queryInterface.addColumn(
+          'events',
+          'mode',
+          {
+            type: Sequelize.STRING,
+            allowNull: true
+          }, { transaction: t })]).then(queryInterface.sequelize.query(`
```

Comment:
Should `category` and `mode` be non-nullable?  When we first added the frontend support we wrote a migration that back-filled existing events, so we _should_ have data for all events.

Reply:
gooood question - i like that idea. this might require further investigation though.

poking in the latest database dump (as it is before running this migration) i see 3927 events total:
- 2231 of these have no category tag
- 1931 of these have no mode tag

[this](https://infinite.industries/events/d8bcbd69-bdf0-4c25-ae71-7fed2cea3f30) is the most recently-created of these events, and it was created on August 25 2022. we have events _without_ category and/or mode going all the way back to 2018. the earliest event _with_ a mode tag was created in August 2019, and the earliest event _with_ a category tag was created in April 2020.

so either we lost some old data in August 2022, or we made a change around that time that ensured that `category` and `mode` would always be stored. 🤔🤔

Reply:
Ah, that's interesting. I thought we had run a migration or script to populate old events with category and mode tags, but maybe I'm misremembering. Or did we accidentally blow that away with the accidental production DB deletion?

Reply:
Based on discussion in today's meeting, and looking back at the initial implementation of mode and category (#289), it looks like we bailed on back-filling these values except for events that had the `remote` and `online-resource` tags from the COVID-era implementation.